core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1578,144 +1409,85 @@ class Log(@volatile private var _dir: File,
         .map(_.messageOffset)
         .getOrElse(maxOffsetInMessages - Integer.MAX_VALUE)
 
-      roll(Some(rollOffset))
+      val newSegment = localLog.roll(Some(rollOffset))
+      afterRoll(newSegment)
+      newSegment
     } else {
       segment
     }
   }
 
   /**
-   * Roll the log over to a new active segment starting with the current logEndOffset.
+   * Roll the local log over to a new active segment starting with the current logEndOffset.
```

Comment:
This comment is not very accurate since we roll to expectedNextOffset or logEndOffset.

Reply:
Sure, I'll fix it. Good catch.

Reply:
Done in 8ebb39f.